lib/docs/query.rb

```diff
@@ -210,15 +210,32 @@ def create_tags_object(tags)
       tags_object.sort_by { |tag| [tag[:active] ? 0 : 1, -tag[:count]] }
     end
 
-    def create_categories_object(categories)
-      categories_object = []
+    def create_categories_object(category_counts)
+      categories =
+        Category
+          .where(id: category_counts.keys)
+          .includes(
+            :uploaded_logo,
+            :uploaded_logo_dark,
+            :uploaded_background,
+            :uploaded_background_dark,
+          )
+          .joins("LEFT JOIN topics t on t.id = categories.topic_id")
+          .select("categories.*, t.slug topic_slug")
 
-      categories.each do |category|
-        active = @filters[:category].include?(category[0].to_s) if @filters[:category]
-        categories_object << { id: category[0], count: category[1], active: active || false }
-      end
+      Category.preload_user_fields!(@guardian, categories)
```

Comment:
I think we need some sort of API in core for people to query/load categories in a performant way since all these includes and preloading of files can be quite involved.

Reply:
I agree, I think we can have a `preload_basic_fields` defined that we can reuse. I will follow up with PRs for core and this plugin for that.